node/src/components/storage.rs

```diff
@@ -1616,86 +1579,194 @@ fn initialize_block_body_v1_db(
             }
         }
     }
+
     txn.commit()?;
     info!("v1 block body database initialized");
     Ok(())
 }
 
-fn delete_merkle_block_body_part(
-    tx: &mut RwTransaction,
-    block_body_v2_db: &Database,
-    value_database: &Database,
-    key: Digest,
-) -> Result<Digest, LmdbExtError> {
-    let [key_to_part_db, merkle_proof_of_rest]: [Digest; 2] =
-        match tx.get_value(*block_body_v2_db, &key)? {
+fn get_merkle_linked_list_node<Tx, T>(
+    tx: &mut Tx,
+    block_body_v2_db: Database,
+    part_database: Database,
+    key_to_block_body_db: &Digest,
+) -> Result<Option<MerkleLinkedListNode<T>>, LmdbExtError>
+where
+    Tx: Transaction,
+    T: DeserializeOwned,
+{
+    let [part_to_value_db, merkle_proof_of_rest]: [Digest; 2] =
+        match tx.get_value(block_body_v2_db, key_to_block_body_db)? {
             Some(slice) => slice,
-            None => {
-                return Err(CouldNotDeleteBlockBodyPart {
-                    merkle_linked_list_node_hash: key,
-                })
-            }
+            None => return Ok(None),
         };
-    tx.del(*value_database, &key_to_part_db, None)?;
-    tx.del(*block_body_v2_db, &key, None)?;
-    Ok(merkle_proof_of_rest)
+    let value = match tx.get_value(part_database, &part_to_value_db)? {
+        Some(value) => value,
+        None => return Ok(None),
+    };
```

Comment:
Shouldn't this be a loop so that it's prepared for versions of the block body with more than 3 parts?

Reply:
What do you mean? This function (`get_merkle_linked_list_node`) only reads a single part of the block, regardless of how many parts there are.

If you mean `get_single_block_body` below, then it would be nice to make it a loop, but I think it's impossible, unfortunately, because we'd have no way of deserializing the parts into correct types in a loop.